### PR TITLE
gdk: Implement Send+Sync for texture

### PR DESCRIPTION
**Pull request by A6GibKm**
From the docs they are immutable and safe
https://docs.gtk.org/gdk4/class.Texture.html

See https://gitlab.gnome.org/GNOME/gtk/-/commit/6785461c2651ad502f46521536619ea77f04dd63

### CODE REVIEW

File: gdk4/src/texture.rs [hunk 1]
```diff
@@ -5,6 +5,9 @@ use crate::Texture;
 use glib::translate::*;
 use glib::IsA;
 
+unsafe impl Send for Texture {}
+unsafe impl Sync for Texture {}
```
Comment (sdroege) [inline gdk4/src/texture.rs:9]: Do something like this instead:

https://github.com/gtk-rs/gtk-rs-core/blob/c53f692d84f7f7a3a75362cd424427842f792e66/gio/Gir.toml#L331

Also the same for all `GdkTexture` subclasses (memory, gl)

Comment (A6GibKm) [reply gdk4/src/texture.rs]: Done, thanks.

**Review by sdroege — APPROVED** [review]

**Comment by sdroege** [comment]
Needs https://github.com/gtk-rs/gtk4-rs

**Comment by bilelmoussaoui** [comment]
Please rebase

**Comment by sdroege** [comment]
Put an `#![allow(clippy::non_send_fields_in_send_ty)]` into `lib.rs` at the top

**Comment by A6GibKm** [comment]
> Put an `#![allow(clippy::non_send_fields_in_send_ty)]` into `lib.rs` at the top

We also have a 
```
#[allow(unknown_lints)]
```

**Comment by bilelmoussaoui** [comment]
> > Put an `#![allow(clippy::non_send_fields_in_send_ty)]` into `lib.rs` at the top
> 
> We also have a
> 
> ```
> #[allow(unknown_lints)]
> ```

could be done there as well no?

**Comment by bilelmoussaoui** [comment]
that easier to remove later once that lint is dropped from clippy

**Pull request merged**